src/main/kotlin/lotto/Calculator.kt

```diff
@@ -0,0 +1,45 @@
+package lotto
+
+class Calculator(
+    tickets: MutableList<Lotto>,
+    val winningNumbers: WinningNumbers,
+    var results: MutableMap<Rank, Int> = emptyMap<Rank, Int>().toMutableMap(),
+) {
+    init {
+        tickets.forEach {
+            val count = findMatches(it, winningNumbers.winningNumbers)
+            val rank =
+                Rank.valueOf(
+                    count,
+                    bonusNumberPresent(it, winningNumbers.bonusNumber),
+                )
+            results[rank] = results.getOrDefault(rank, 0) + 1
+        }
+    }
+
+    fun calculateReturnRate(purchaseAmount: Int): Float {
+        return (calculateTotalEarnings() / purchaseAmount.toFloat())
+    }
+
+    fun calculateTotalEarnings(): Float {
```

Comment:
I recommend exploring the advantages of method chaining and trying to refactor your code accordingly 😉
Kotlin provides a variety of built-in functions to support this style.